# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/libcamera/hotfix-clone3-ia64.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- libcamera.git/src/libcamera/process.cpp.vanilla	2026-02-28 15:52:45.304466488 +0100
+++ libcamera.git/src/libcamera/process.cpp	2026-02-28 15:56:12.722226991 +0100
@@ -157,7 +157,11 @@
 	cargs.pidfd = reinterpret_cast<uintptr_t>(&pidfd);
 	cargs.exit_signal = SIGCHLD;
 
+#ifdef __ia64
+	long childPid = syscall(SYS_clone2, &cargs, sizeof(cargs));
+#else
 	long childPid = syscall(SYS_clone3, &cargs, sizeof(cargs));
+#endif
 	if (childPid < 0) {
 		int ret = -errno;
 		LOG(Process, Error) << "Failed to fork: " << strerror(-ret);
